### PR TITLE
Mc 3733 add public ip service

### PR DESCRIPTION
**Pull request by franzpgarcia**

### CODE REVIEW

**Comment by pdube** [comment]
+1 for go fmt

File: services/cloudca/public_ip.go [hunk 1]
```diff
@@ -1,8 +1,110 @@
...
-	Ports []string `json:"ports,omitempty"`
-}
\ No newline at end of file
+	Id            string   `json:"id,omitempty"`
+	IpAddress     string   `json:"ipaddress,omitempty"`
+	State         string   `json:"state,omitempty"`
+	ZoneId        string   `json:"zoneId,omitempty"`
+	ZoneName      string   `json:"zoneName,omitempty"`
+	NetworkId     string   `json:"networkId,omitempty"`
+	Network       string   `json:"network,omitempty"`
```
Comment (pdube) [inline services/cloudca/public_ip.go]: Network name?

**Comment by pdube** [comment]
👍

**Comment by simongodard** [comment]
I know you guys already discussed it last week, but I wonder if we should drop support for names when linking resources? You typically don't hardcode names and IDs directly in your Terraform file anyways...

File: services/cloudca/public_ip.go (hunk 1 above)
Comment (franzpgarcia) [inline services/cloudca/public_ip.go]: Done

**Comment by pdube** [comment]
I would be fine with dropping name support. I am not a big fan of it anyways

**Pull request merged**